databricks_cli/secrets/cli.py

```diff
@@ -0,0 +1,308 @@
+        content = click.edit(prompt)
+        # return None means editor is closed without changes
+        if content is None:
+            error_and_quit('No changes made, write secret aborted.'
+                           ' Please follow the instruction to input secret value.')
+
+        return content.rstrip('\n'), None
+
+    if string_value is not None:
+        return string_value, None
+
+    if binary_file is not None:
+        with open(binary_file, 'rb') as f:
+            binary_content = f.read().rstrip('\n')
+
+        if not is_base64_str(binary_content):
+            error_and_quit("Content in file {} is probably not base64 encoded.".format(binary_file))
+
+        return None, binary_content
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Writes a secret to a scope.')
+@click.option('--scope', required=True)
+@click.option('--key', required=True)
+@click.option('--string-value', default=None,
+              help='Read value from string and stored in UTF-8 (MB4) form')
+@click.option('--binary-file', default=None, type=click.Path(exists=True, readable=True),
+              help='Read value from binary-file and stored as bytes.')
+@profile_option
+@eat_exceptions
+@provide_api_client
+def write_secret(api_client, scope, key, string_value, binary_file):
+    """
+    Writes a secret to the provided scope with the given name. Overwrites if the name exists.
+
+    You should specify at most one option in "string-value" and "binary-file".
+
+    If "string-value", the argument will be stored in UTF-8 (MB4) form.
+
+    If "binary-file", the argument should be a path to file. The file should store
+    base64 encoded data. File content will be read as secret value and stored as bytes.
+
+    If none of "string-value" and "binary-file" specified, an editor will be opened for
+    inputting secret value. The value will be stored in UTF-8 (MB4) form.
+    """
+    string_param, bytes_param = _verify_and_translate_options(string_value, binary_file)
+    SecretApi(api_client).write_secret(scope, key, string_param, bytes_param)
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Deletes a secret.')
+@click.option('--scope', required=True)
+@click.option('--key', required=True)
+@profile_option
+@eat_exceptions
+@provide_api_client
+def delete_secret(api_client, scope, key):
+    """
+    Deletes the secret stored in this scope.
+    """
+    SecretApi(api_client).delete_secret(scope, key)
+
+
+def _secrets_to_table(secrets_json):
+    ret = []
+    for s in secrets_json.get('secrets', []):
+        ret.append((s['key'], s.get('last_updated_timestamp', 'Not Available')))
+    return ret
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Lists all the secrets in a scope.')
+@click.option('--scope', required=True)
+@click.option('--output', default=None, help=OutputClickType.help, type=OutputClickType())
```

Comment:
set default here to whatever we agreed to in the doc

Reply:
same for other `--output` options

Reply:
The default value is handled in `OutputClickType` and click's default is already `None` if not specified. I am removing `default=None` for now.